File: backend-java/src/main/java/com/aiproject/health/HealthController.java
package com.aiproject.health;

import com.aiproject.common.ApiResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.connection.RedisConnection;
import org.springframework.data.redis.connection.RedisConnectionFactory;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.Instant;
import java.util.LinkedHashMap;
import java.util.Map;

/**
 * 健康检查：服务状态 + 数据库 + Redis 连通性。
 */
@RestController
@RequestMapping("/api/v1/health")
@RequiredArgsConstructor
public class HealthController {

    private final JdbcTemplate jdbcTemplate;
    private final RedisConnectionFactory redisConnectionFactory;

    @GetMapping
    public ApiResponse<Map<String, Object>> health() {
        Map<String, Object> data = new LinkedHashMap<>();
        data.put("status", "UP");
        data.put("service", "backend-java");
        data.put("db", checkDb());
        data.put("redis", checkRedis());
        data.put("timestamp", Instant.now().toString());
        return ApiResponse.ok(data);
    }

    private String checkDb() {
        try {
            jdbcTemplate.queryForObject("SELECT 1", Integer.class);
            return "UP";
        } catch (Exception e) {
            return "DOWN";
        }
    }

    private String checkRedis() {
        try (RedisConnection connection = redisConnectionFactory.getConnection()) {
            connection.ping();
            return "UP";
        } catch (Exception e) {
            return "DOWN";
        }
    }
}

File: backend-dotnet/AiProject.Api/Controllers/HealthController.cs
using AiProject.Api.Common;
using AiProject.Api.Data;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace AiProject.Api.Controllers;

[ApiController]
[Route("api/v1/health")]
public class HealthController : ControllerBase
{
    private readonly AppDbContext _db;

    public HealthController(AppDbContext db)
    {
        _db = db;
    }

    [HttpGet]
    [AllowAnonymous]
    public async Task<ActionResult<ApiResponse<object>>> Health()
    {
        string dbStatus = "UP";
        try
        {
            await _db.Database.ExecuteSqlRawAsync("SELECT 1");
        }
        catch
        {
            dbStatus = "DOWN";
        }

        return ApiResponse<object>.Ok(new
        {
            status = "UP",
            service = "backend-dotnet",
            db = dbStatus,
            timestamp = DateTime.UtcNow.ToString("o")
        });
    }
}
